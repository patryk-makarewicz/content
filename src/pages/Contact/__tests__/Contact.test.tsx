import { render } from '@testing-library/react';
import React from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { RecoilRoot } from 'recoil';

import { Contact } from '../Contact';

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

const renderContact = () => {
  const { asFragment } = render(
    <RecoilRoot>
      <QueryClientProvider client={queryClient}>
        <Contact />
      </QueryClientProvider>
    </RecoilRoot>
  );

  return {
    asFragment,
  };
};

describe('test Contact', () => {
  it('should take and matches the snapshot', () => {
    const { asFragment } = renderContact();
    const firstRender = asFragment();

    expect(firstRender).toMatchSnapshot();
  });
});
