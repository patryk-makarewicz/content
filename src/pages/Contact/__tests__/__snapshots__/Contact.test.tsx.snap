// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Contact should take and matches the snapshot 1`] = `
<DocumentFragment>
  <section
    class="sc-jlZhew fVCUvx"
  >
    <h2
      class="sc-dAlyuH fwxLVM"
      data-testid="section-title"
    >
      contact.title
    </h2>
    <p
      class="sc-cwHptR izyYR"
    >
      contact.description
    </p>
    <div
      class="sc-jEACwC jzzNTX"
    >
      <form>
        <div
          class="sc-fqkvVR cWzMDi"
        >
          <label
            class="sc-gsFSXq lWbfn"
            for="firstName"
          >
            contact.firstName
          </label>
          <input
            class="sc-kAyceB hBqCkZ"
            id="firstName"
            maxlength="70"
            name="firstName"
            placeholder="contact.firstName"
            type="text"
          />
        </div>
        <div
          class="sc-fqkvVR cWzMDi"
        >
          <label
            class="sc-gsFSXq lWbfn"
            for="lastName"
          >
            contact.lastName
          </label>
          <input
            class="sc-kAyceB hBqCkZ"
            id="lastName"
            maxlength="70"
            name="lastName"
            placeholder="contact.lastName"
            type="text"
            value=""
          />
        </div>
        <div
          class="sc-fqkvVR cWzMDi"
        >
          <label
            class="sc-gsFSXq lWbfn"
            for="email"
          >
            contact.email
          </label>
          <input
            class="sc-kAyceB hBqCkZ"
            id="email"
            maxlength="70"
            name="email"
            placeholder="contact.email"
            type="email"
          />
        </div>
        <div
          class="sc-fqkvVR cWzMDi"
        >
          <label
            class="sc-gsFSXq lWbfn"
            for="phone"
          >
            contact.phone
          </label>
          <input
            class="sc-kAyceB hBqCkZ"
            id="phone"
            maxlength="70"
            name="phone"
            placeholder="contact.phone"
            type="tel"
          />
        </div>
        <div
          class="sc-fqkvVR cWzMDi"
        >
          <label
            class="sc-gsFSXq lWbfn"
            for="message"
          >
            contact.message
          </label>
          <textarea
            class="sc-imWYAI hTQPsO"
            id="message"
            maxlength="1000"
            name="message"
            placeholder="contact.messagePlaceholder"
          />
        </div>
        <div
          style="display: flex; justify-content: center;"
        >
          <button
            class="sc-eqUAAy fOohoi"
            type="submit"
          >
            contact.submit
          </button>
        </div>
      </form>
    </div>
    <div
      class="sc-dhKdcB dMHnuT"
    >
      <a
        class="sc-kpDqfm exMwnz"
        href="https://www.linkedin.com/in/patryk-makarewicz/"
        target="_blank"
        title="Go to LinkedIn"
      >
        <img
          alt="LinkedIn logo"
          height="36"
          src="[object Object]"
          width="36"
        />
      </a>
      <a
        class="sc-kpDqfm exMwnz"
        href="https://github.com/patryk-makarewicz"
        target="_blank"
        title="Go to Github"
      >
        <img
          alt="Github logo"
          height="36"
          src="[object Object]"
          width="36"
        />
      </a>
    </div>
  </section>
</DocumentFragment>
`;
